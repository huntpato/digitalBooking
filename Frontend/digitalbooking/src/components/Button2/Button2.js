import React from "react";
import styles from "./button2.module.css";
import { Link } from "react-router-dom"

export const Button2 = (props) => {
  const { text, link, handleClick } = props;
  const { button2 } = styles;
  
  return (
    <Link to= {link || '/'} style={{textDecoration: 'none'}} onClick={ handleClick|| ''}> <button className={button2}>
    {text}</button></Link> 
  );
};
