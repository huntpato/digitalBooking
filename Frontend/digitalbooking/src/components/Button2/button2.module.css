.button2 {
  align-items: center;
  background-color: var(--principal-color);
  border: 1.4px solid var(--background-white);
  border: none;
  border-radius: 5px;
  color: white;
  cursor: pointer;
  display: flex;
  justify-content: center;
  font-weight: 700;
  height: 32px;
  margin-left: 0px;
  padding: 6px 40px 5px 40px;
  text-decoration: none;
  transition: 0.5s;
  width: 100%;
  z-index: 0;
}

.button2:active {
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.15);
  transform: scale(0.97);
  
}

@media (max-width: 768px){

  .button2{
    padding: 6px 30px 5px 30px;
  }

}
