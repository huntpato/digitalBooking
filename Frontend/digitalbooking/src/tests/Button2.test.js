import React from 'react';
import { shallow } from 'enzyme';
import { Button2 } from '../components/Button2/Button2'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals';

describe('Pruebas en componente <Button2 />', ()=> {
    const text = "Ver mas";
    const wrapper= shallow(<Button2 text={text}/>);

    test('Comprobar que <Button2 /> renderiza correctamente.', () => {
    
        expect( wrapper ).toMatchSnapshot();
      
    })

    test('Verificar que <Button2 /> muestra el texto enviado por props', () => {
    
        const button = wrapper.find('Link').text();
        expect( button ).toContain( text );
    }) 

    test('Verificar que el componente carga la clase de CSS correctamente', () => {

        const button = wrapper.find('button');
        expect(button.hasClass('button2')).toBe(true);
    })
    

})
