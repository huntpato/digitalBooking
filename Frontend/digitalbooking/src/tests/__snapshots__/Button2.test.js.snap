// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <Button2 /> Comprobar que <Button2 /> renderiza correctamente. 1`] = `
<Link
  onClick=""
  style={
    Object {
      "textDecoration": "none",
    }
  }
  to="/"
>
   
  <button
    className="button2"
  >
    Ver mas
  </button>
</Link>
`;
